clang/lib/Driver/Compilation.cpp

```diff
@@ -126,6 +127,21 @@ bool Compilation::CleanupFile(const char *File, bool IssueErrors) const {
   // Don't try to remove files which we don't have write access to (but may be
   // able to remove), or non-regular files. Underlying tools may have
   // intentionally not overwritten them.
+
+  // If -fsycl-dump-device-code option is enabled, filter .spv files
+  // from the list of temporary files and save them in the user provided
+  // directory.
+  std::string FileNameExt;
+  std::string FileName = File;
+  if (TheDriver.isDumpDeviceCodeEnabled()) {
+    size_t PositionOfDot = FileName.rfind('.', FileName.length());
+    if (PositionOfDot != std::string::npos)
+      FileNameExt =
+          FileName.substr(PositionOfDot + 1, FileName.length() - PositionOfDot);
+  }
+  if (FileNameExt == "spv")
+    return false;
+
```

Comment:
I'm not following how this is still needed.  If you are peeling off the 'Code' in the table, that includes all of the .spv files.  This means no .spv files from that table should be going through `CleanupFile`.  This check here is also general, and impacts all .spv files seen during the cleanup.

Reply:
>> If you are peeling off the 'Code' in the table, that includes all of the .spv files. This means no .spv files from that table should be going through CleanupFile.

Before peeling:
[Code|Properties|Symbols]
/iusers/sundaram//sycl-dump-device-code-337093-fa4200.spv|/tmp/sycl-dump-device-code-e973e9_0.prop|/tmp/sycl-dump-device-code-e973e9_0.sym

After peeling:
[Code|Properties|Symbols]
/iusers/sundaram//sycl-dump-device-code-337093-fa4200.spv

Peeling "Code" column removes all columns except "Code" column.

And after peeling, each of the files in the table are passed to CleanupFile.